www/apps/book/app/advanced-development/modules/db-operations/page.mdx

```diff
@@ -0,0 +1,482 @@
+
+If your transactional method uses other methods that accept a Medusa context, pass the shared context to those method.
+
+For example:
+
+```ts
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
+  protected async anotherMethod(
+    @MedusaContext() sharedContext?: Context<EntityManager>
+  ) {
+    // ...
+  }
+  
+  @InjectTransactionManager()
+  protected async update_(
+    input: {
+      id: string,
+      name: string
+    },
+    @MedusaContext() sharedContext?: Context<EntityManager>
+  ): Promise<any> {
+    anotherMethod(sharedContext)
+  }
+}
+```
+
+You use the `anotherMethod` transactional method in the `update_` transactional method, so you pass it the shared context.
+
+The `anotherMethod` now runs in the same transaction as the `update_` method.
+
+---
+
+## Configure Transactions
+
+To configure the transaction, such as its [isolation level](https://www.postgresql.org/docs/current/transaction-iso.html), use the `baseRepository` dependency registered in your module's container.
+
+The `baseRepository` is an instance of a repository class that provides methods to create transactions, run database operations, and more.
+
+The `baseRepository` has a `transaction` method that allows you to run a function within a transaction and configure that transaction.
+
+For example, resolve the `baseRepository` in your service's constructor:
+
+<CodeTabs group="service-type">
+  <CodeTab label="Extending Service Factory" value="service-factory">
+
+```ts highlights={[["14"]]}
+import { MedusaService } from "@medusajs/utils"
+import MyCustom from "./models/my-custom"
+import { DAL } from "@medusajs/types"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+}
+
+class HelloModuleService extends MedusaService({
+  MyCustom,
+}){
+  protected baseRepository_: DAL.RepositoryService
+
+  constructor({ baseRepository }: InjectedDependencies) {
+    super(...arguments)
+    this.baseRepository_ = baseRepository
+  }
+}
+
+export default HelloModuleService
+```
+  </CodeTab>
+  <CodeTab label="Without Service Factory" value="no-service-factory">
+
+```ts highlights={[["10"]]}
+import { DAL } from "@medusajs/types"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+}
+
+class HelloModuleService {
+  protected baseRepository_: DAL.RepositoryService
+
+  constructor({ manager }: InjectedDependencies) {
+    this.baseRepository_ = baseRepository
+  }
+}
+
+export default HelloModuleService
+```
+
+  </CodeTab>
+</CodeTabs>
+
+Then, add the following method that uses it:
+
+export const repoHighlights = [
+  ["20", "transaction", "Wrap the function parameter in a transaction."]
+]
+
+```ts highlights={repoHighlights}
+// ...
+import { 
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext
+} from "@medusajs/utils"
+import { Context } from "@medusajs/types"
+import { EntityManager } from "@mikro-orm/knex"
+
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
```

Comment:
here we dont need this decorator since we are creating a transaction in the method body right? is there any example purpose to mix both here, also if there is a need, then we should mention passing the transaction manager of the method to the transaction options in the body

Reply:
This was part of our conversation yesterday on whether to always include the decorator. Then the fix here would be to pass the transaction to the `transaction` method, right?